File: Common/InMotionGIT.Common - Copy/DataAcceess/WhereStatements/NumericCondition.vb
Namespace DataAccess.WhereStatements

    Public Class NumericCondition(Of t)

        Private ReadOnly fieldName As String
        Private ReadOnly tableType As IWhereStatement

        Public Sub New(name As String, base As IWhereStatement)
            fieldName = name
            tableType = base
        End Sub

#Region "EqualTo"

        Public Function EqualTo(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} = {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function EqualTo(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} = {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function EqualTo(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} = {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "NotEqualTo"

        Public Function NotEqualTo(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <> {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function NotEqualTo(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <> {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function NotEqualTo(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <> {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "GreaterThan"

        Public Function GreaterThan(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} > {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThan(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} > {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThan(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} > {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "LessThan"

        Public Function LessThan(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} < {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThan(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} < {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThan(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} < {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "GreaterThanEqualTo"

        Public Function GreaterThanEqualTo(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} >= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThanEqualTo(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} >= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThanEqualTo(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} >= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "LessThanEqualTo"

        Public Function LessThanEqualTo(value As Integer) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThanEqualTo(value As Double) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThanEqualTo(value As Decimal) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <= {1}", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "IsNull"

        Public Function IsNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

#Region "IsNotNull"

        Public Function IsNotNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NOT NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

#End Region

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/NumericCondition.cs

namespace InMotionGIT.Common.DataAccess.WhereStatements
{

    public class NumericCondition<t>
    {

        private readonly string fieldName;
        private readonly IWhereStatement tableType;

        public NumericCondition(string name, IWhereStatement @base)
        {
            fieldName = name;
            tableType = @base;
        }

        #region EqualTo

        public LogicalOperator<t> EqualTo(int value)
        {
            tableType.command += string.Format("{0} = {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> EqualTo(double value)
        {
            tableType.command += string.Format("{0} = {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> EqualTo(decimal value)
        {
            tableType.command += string.Format("{0} = {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region NotEqualTo

        public LogicalOperator<t> NotEqualTo(int value)
        {
            tableType.command += string.Format("{0} <> {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> NotEqualTo(double value)
        {
            tableType.command += string.Format("{0} <> {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> NotEqualTo(decimal value)
        {
            tableType.command += string.Format("{0} <> {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region GreaterThan

        public LogicalOperator<t> GreaterThan(int value)
        {
            tableType.command += string.Format("{0} > {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThan(double value)
        {
            tableType.command += string.Format("{0} > {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThan(decimal value)
        {
            tableType.command += string.Format("{0} > {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region LessThan

        public LogicalOperator<t> LessThan(int value)
        {
            tableType.command += string.Format("{0} < {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThan(double value)
        {
            tableType.command += string.Format("{0} < {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThan(decimal value)
        {
            tableType.command += string.Format("{0} < {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region GreaterThanEqualTo

        public LogicalOperator<t> GreaterThanEqualTo(int value)
        {
            tableType.command += string.Format("{0} >= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThanEqualTo(double value)
        {
            tableType.command += string.Format("{0} >= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThanEqualTo(decimal value)
        {
            tableType.command += string.Format("{0} >= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region LessThanEqualTo

        public LogicalOperator<t> LessThanEqualTo(int value)
        {
            tableType.command += string.Format("{0} <= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThanEqualTo(double value)
        {
            tableType.command += string.Format("{0} <= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThanEqualTo(decimal value)
        {
            tableType.command += string.Format("{0} <= {1}", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region IsNull

        public LogicalOperator<t> IsNull()
        {
            tableType.command += string.Format("{0} IS NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

        #region IsNotNull

        public LogicalOperator<t> IsNotNull()
        {
            tableType.command += string.Format("{0} IS NOT NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        #endregion

    }

}